### PR TITLE
[issue-705]Fix data cleanup before data load and data load failure

### PR DESCRIPTION
**Pull request by eason-lyx**
link issue #705 
this pr to fix data cleanup ,using new file structure  to replace old file structure which is no use.

### CODE REVIEW

File: integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java [hunk 1]
```diff
@@ -271,43 +286,66 @@ public static void deleteSlice(int partitionCount, String schemaName, String cub
...
+    String tableStatusPath =
+        loadModel.getCarbonDataLoadSchema().getCarbonTable().getMetaDataFilepath() + File.separator
+        + CarbonCommonConstants.LOADMETADATA_FILENAME;
+    Gson gsonObjectToRead = new Gson();
+    DataInputStream dataInputStream = null;
+    LoadMetadataDetails[] listOfLoadFolderDetailsArray = null;
+    try {
+      if (FileFactory.isFileExist(tableStatusPath, FileFactory.getFileType(tableStatusPath))) {
+        dataInputStream = FileFactory
+          .getDataInputStream(tableStatusPath, FileFactory.getFileType(tableStatusPath));
```
Comment (ravikiran23) [inline integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java]: please use SegmentStatusManager to read the table status file. Already API is present for this purpose.

File: integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java [hunk 2]
```diff
@@ -271,43 +286,66 @@ public static void deleteSlice(int partitionCount, String schemaName, String cub
...
+          CarbonFile[] listFiles = carbonFile.listFiles(new CarbonFileFilter() {
+            @Override
+            public boolean accept(CarbonFile path) {
+              return !loadFolders.contains(path.getAbsolutePath().replace("\\", "/"));
+            }
+          });
+          for (int k = 0; k < listFiles.length; k++) {
+            deleteStorePath(listFiles[k].getAbsolutePath());
+          }
+        }
```
Comment (manishgupta88) [inline integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java]: @eason-lyx ...can  you please try to split this method into 2 methods....

File: integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java [hunk 3]
```diff
@@ -271,43 +287,39 @@ public static void deleteSlice(int partitionCount, String schemaName, String cub
...
-    String partitionSchemaName = null;
-    String partitionCubeName = null;
-    for (int i = 0; i < partitionCount; i++) {
-      partitionSchemaName = schemaName + '_' + i;
-      partitionCubeName = cubeName + '_' + i;
-      tableLoc =
-          getTableLocation(partitionSchemaName, partitionCubeName, tableName, hdfsStoreLocation,
-              currentRestructNumber);
+  public static void deletePartialLoadDataIfExist(CarbonLoadModel loadModel) throws IOException {
+    CarbonTable carbonTable = org.carbondata.core.carbon.metadata.CarbonMetadata.getInstance()
```
Comment (QiangCai) [inline integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java]: you can get carbonTable from loadModel.

**Comment by QiangCai** [comment]
@eason-lyx  please check "delete segment" function. I found it cannot delete segment folder.

**Comment by eason-lyx** [comment]
@QiangCai delete segments will only change status of segment folder in tablestatus file. It will change the status to "Marked for delete". It will not delete the segment.

**Comment by QiangCai** [comment]
LGTM

**Pull request merged**